apps/voting/contracts/Voting.sol

```diff
@@ -519,40 +545,35 @@ contract Voting is IForwarder, AragonApp {
      * @dev Internal function to cast a vote or object to.
      * @dev It assumes that voter can support or object to the vote
      * @param _voteId The identifier of the vote
-     * @param _supports Whether the voter supports the vote or not
+     * @param _supports Whether the voter supports the vote
      * @param _voter The address of the voter
-     * @param _isDelegate Whether the voter is a delegate or not
+     * @param _isDelegate Whether the voter is a delegate
      */
-    function _vote(uint256 _voteId, bool _supports, address _voter, bool _isDelegate) internal {
+    function _vote(uint256 _voteId, bool _supports, address _voter, bool _isDelegate, uint256 _votingPower) internal {
         Vote storage vote_ = votes[_voteId];
-
-        // This could re-enter, though we can assume the governance token is not malicious
-        uint256 voterStake = token.balanceOfAt(_voter, vote_.snapshotBlock);
         VoterState state = vote_.voters[_voter];
 
         // If voter had previously voted, decrease count
-        if (state == VoterState.Yea || state == VoterState.DelegateYea) {
-            vote_.yea = vote_.yea.sub(voterStake);
-        } else if (state == VoterState.Nay || state == VoterState.DelegateNay) {
-            vote_.nay = vote_.nay.sub(voterStake);
+        if (state != VoterState.Absent) {
```

Comment:
Please add a comment noting that this check is a gas optimization, and minimizes validations for the most common case where voters cast their votes once.

Reply:
Addressed in f1f6de1606a6eda1999a0e1cc0ea895857101f88.